### PR TITLE
fix: modules duplicated in multiple chunks now are always loaded by runtime.js

### PR DESCRIPTION
**Pull request by maunier**
## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Here is the issue: https://github.com/vuejs/vue/issues/11560
Sorry I have maken the issue at wrong place.

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by clarkdo — COMMENTED** [review]
Is the only change removing `m.chunks.length === 1` ?

**Comment by maunier** [comment]
> Is the only change removing `m.chunks.length === 1` ?

Yes

**Comment by clarkdo** [comment]
This plugin is cloned from vue, so would mind opening an issue there ? https://github.com/vuejs/vue/blob/4dec3b52c9b71f816e6b86d42ea53e9f2e559646/src/server/webpack-plugin/client.js#L40

I think introducing multiple chunks may need more analyse and data processing, I'll take a deep look about the change.

**Comment by maunier** [comment]
> This plugin is cloned from vue, so would you mind opening an issue there ? https://github.com/vuejs/vue/blob/4dec3b52c9b71f816e6b86d42ea53e9f2e559646/src/server/webpack-plugin/client.js#L40
> 
> I think introducing multiple chunks may need more analyse and data processing, I'll take a deep look about the change.

ok, thanks

**Comment by clarkdo** [comment]
BTW, can you please provide me a minimal reproduciable repo about this issue ?

**Comment by maunier** [comment]
> BTW, can you please provide me a minimal reproduciable repo about this issue ?

Yes: https://github.com/maunier/test-vue-ssr.git

File: packages/webpack/src/plugins/vue/client.js [hunk 1]
```diff
@@ -55,48 +55,45 @@ export default class VueSSRClientPlugin {
...
-          // Find all asset modules associated with the same chunk
-          assetModules.forEach((m) => {
-            if (m.chunks.some(id => id === cid)) {
-              files.push.apply(files, m.assets.map(fileToIndex))
-            }
-          })
         }
+
+        // Find all asset modules associated with the same chunk
+        assetModules.forEach((m) => {
```
Comment (galvez) [inline packages/webpack/src/plugins/vue/client.js:92]: @clarkdo Any reason not to use `for..of` here, to keep consistency with the rest of the code?

**Comment by pi0** [comment]
@clarkdo is investigating more but it is probably not going to merged.

**Comment by pi0** [comment]
@clarkdo is investigating more but it is probably not going to merged. BTW thanks for PR @maunier heart

**Review by Kronhyx — APPROVED** [review]

**Review by rahulkumarsingh73690 — APPROVED** [review]

**Comment by pi0** [comment]
Hi @maunier Testing repro (https://github.com/maunier/test-vue-ssr.git) with fresh lock-file seems everything is right. 

chunks:

```
../server/client.manifest.json   6.31 KiB          [emitted]              
                    020e051.js  370 bytes    3, 5  [emitted] [immutable]  pages/c2
                    54aff81.js    155 KiB       1  [emitted] [immutable]  commons/app
                    638c457.js   49.5 KiB       0  [emitted] [immutable]  app
                    71a0567.js  478 bytes       2  [emitted] [immutable]  pages/c1
                      LICENSES  389 bytes          [emitted]              
                    cb2fd46.js   2.32 KiB       4  [emitted] [immutable]  runtime
                    eebef40.js  334 bytes       5  [emitted] [immutable]  
```

`client.manifest.json`:

```js
  "async": [
    "020e051.js", <-- pages/c2
    "71a0567.js", <-- pages/c1
    "eebef40.js"
  ],
```

`server.manifest.json`:

```js
{
  "entry": "server.js",
  "files": {
    "3.js": "3.js",
    "pages/c1.js": "pages/c1.js",
    "pages/c2.js": "pages/c2.js",
    "server.js": "server.js"
  },
  "maps": {
    "3.js": "3.js.map",
    "pages/c1.js": "pages/c1.js.map",
    "pages/c2.js": "pages/c2.js.map",
    "server.js": "server.js.map"
  }
}
```

Also pages are preloaded when locally testing.

I believe this fixed with [2.14.5](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.5)  / #8012

**Pull request closed without merging**